### ISSUE TITLE
💅 noUnnecessaryConditions consistently reporting switch cases as unreachable

### ISSUE THREAD

**Issue by lewiswolf** [biomejs/biome#10633]
### Environment information

I just updated to the latest 2.5 release, and now all switch statements within my codebase are being flagged as having unreachable cases. All of these switch statements are accurately typed, and perform correctly in production. I have included a novel example below.

<details>

```ts
	switch (Math.round(Math.random() * 10)) {
		case 1:
			break
		case 2:
			break
		case 3:
			break
		case 4:
			break
		case 5:
			break
		case 6:
			break
		case 7:
			break
		case 8:
			break
		case 9:
			break
		default:
			break
	}
```

```bash

src/components/App.tsx:41:8 lint/suspicious/noUnnecessaryConditions ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ This case is unreachable.
  
    40 │        switch (Math.round(Math.random() * 10)) {
  > 41 │                case 1:
       │                     ^
    42 │                        break
    43 │                case 2:
  
  ℹ The value passed to switch can never equal this value.
  

src/components/App.tsx:43:8 lint/suspicious/noUnnecessaryConditions ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ This case is unreachable.
  
    41 │                case 1:
    42 │                        break
  > 43 │                case 2:
       │                     ^
    44 │                        break
    45 │                case 3:
  
  ℹ The value passed to switch can never equal this value.
  

src/components/App.tsx:45:8 lint/suspicious/noUnnecessaryConditions ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ This case is unreachable.
  
    43 │                case 2:
    44 │                        break
  > 45 │                case 3:
       │                     ^
    46 │                        break
    47 │                case 4:
  
  ℹ The value passed to switch can never equal this value.
  

src/components/App.tsx:47:8 lint/suspicious/noUnnecessaryConditions ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ This case is unreachable.
  
    45 │                case 3:
    46 │                        break
  > 47 │                case 4:
       │                     ^
    48 │                        break
    49 │                case 5:
  
  ℹ The value passed to switch can never equal this value.
  

src/components/App.tsx:49:8 lint/suspicious/noUnnecessaryConditions ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ This case is unreachable.
  
    47 │                case 4:
    48 │                        break
  > 49 │                case 5:
       │                     ^
    50 │                        break
    51 │                case 6:
  
  ℹ The value passed to switch can never equal this value.
  

src/components/App.tsx:51:8 lint/suspicious/noUnnecessaryConditions ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ This case is unreachable.
  
    49 │                case 5:
    50 │                        break
  > 51 │                case 6:
       │                     ^
    52 │                        break
    53 │                case 7:
  
  ℹ The value passed to switch can never equal this value.
  

src/components/App.tsx:53:8 lint/suspicious/noUnnecessaryConditions ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ This case is unreachable.
  
    51 │                case 6:
    52 │                        break
  > 53 │                case 7:
       │                     ^
    54 │                        break
    55 │                case 8:
  
  ℹ The value passed to switch can never equal this value.
  

src/components/App.tsx:55:8 lint/suspicious/noUnnecessaryConditions ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ This case is unreachable.
  
    53 │                case 7:
    54 │                        break
  > 55 │                case 8:
       │                     ^
    56 │                        break
    57 │                case 9:
  
  ℹ The value passed to switch can never equal this value.
  

src/components/App.tsx:57:8 lint/suspicious/noUnnecessaryConditions ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ This case is unreachable.
  
    55 │                case 8:
    56 │                        break
  > 57 │                case 9:
       │                     ^
    58 │                        break
    59 │                default:
  
  ℹ The value passed to switch can never equal this value.

```

</details>

Thank you for all your hard work also! I love biome, and I'm very excited for the scss updates coming soon! <3

### Rule name

lint/suspicious/noUnnecessaryConditions 

### Playground link

[valid playground link](https://biomejs.dev/playground/?lintRules=noUnnecessaryConditions&tab=syntax&pane=Console&code=YwBvAG4AcwB0ACAAcwBvAG0AZQBfAHMAdAByAGkAbgBnADoAIABzAHQAcgBpAG4AZwAgAD0AIAAnAGEAbgBvAHQAaABlAHIALQB0AGUAcwB0ACcACgAKAHMAdwBpAHQAYwBoACAAKABzAG8AbQBlAF8AcwB0AHIAaQBuAGcAKQAgAHsACgAgACAAIAAgAGMAYQBzAGUAIAAnAHQAZQBzAHQAJwA6AAoAIAAgACAAIAAgACAAIAAgAGIAcgBlAGEAawAKACAAIAAgACAAYwBhAHMAZQAgACcAYQBuAG8AdABoAGUAcgAtAHQAZQBzAHQAJwA6AAoAIAAgACAAIAAgACAAIAAgAGMAbwBuAHMAbwBsAGUALgBsAG8AZwAoACcAcwB1AGMAYwBlAHMAcwAnACkACgAgACAAIAAgACAAIAAgACAAYgByAGUAYQBrAAoAIAAgACAAIABkAGUAZgBhAHUAbAB0ADoACgAgACAAIAAgACAAIAAgACAAYgByAGUAYQBrAAoAfQAKAAoAcwB3AGkAdABjAGgAIAAoAE0AYQB0AGgALgByAG8AdQBuAGQAKABNAGEAdABoAC4AcgBhAG4AZABvAG0AKAApACAAKgAgADEAMAApACkAIAB7AAoAIAAgACAAIABjAGEAcwBlACAAMQA6AAoAIAAgACAAIAAgACAAIAAgAGIAcgBlAGEAawAKACAAIAAgACAAYwBhAHMAZQAgADIAOgAKACAAIAAgACAAIAAgACAAIABiAHIAZQBhAGsACgAgACAAIAAgAGMAYQBzAGUAIAAzADoACgAgACAAIAAgACAAIAAgACAAYgByAGUAYQBrAAoAIAAgACAAIABjAGEAcwBlACAANAA6AAoAIAAgACAAIAAgACAAIAAgAGIAcgBlAGEAawAKACAAIAAgACAAYwBhAHMAZQAgADUAOgAKACAAIAAgACAAIAAgACAAIABiAHIAZQBhAGsACgAgACAAIAAgAGMAYQBzAGUAIAA2ADoACgAgACAAIAAgACAAIAAgACAAYgByAGUAYQBrAAoAIAAgACAAIABjAGEAcwBlACAANwA6AAoAIAAgACAAIAAgACAAIAAgAGIAcgBlAGEAawAKACAAIAAgACAAYwBhAHMAZQAgADgAOgAKACAAIAAgACAAIAAgACAAIABiAHIAZQBhAGsACgAgACAAIAAgAGMAYQBzAGUAIAA5ADoACgAgACAAIAAgACAAIAAgACAAYgByAGUAYQBrAAoAIAAgACAAIABkAGUAZgBhAHUAbAB0ADoACgAgACAAIAAgACAAIAAgACAAYgByAGUAYQBrAAoAfQA%3D)

### Expected result

no errors reported

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by biome-cookie** [comment]
Hello @lewiswolf, please provide a minimal reproduction. You can use one of the following options:

- Provide a link to [our playground](https://biomejs.dev/playground), if it's applicable.
- Provide a link to GitHub repository. To easily create a reproduction, you can use our interactive CLI via `npm create @biomejs/biome-reproduction`

Issues marked with `S-Needs repro` will be **closed** if they have **no activity within 3 days**.

**Comment by lewiswolf** [comment]
I am not getting the same diagnostics in the playground - I'm unsure how to proceed? 

[My playground example is linked here though!](https://biomejs.dev/playground/?tab=types-ir&pane=Diagnostics&code=YwBvAG4AcwB0ACAAcwBvAG0AZQBfAHMAdAByAGkAbgBnADoAIABzAHQAcgBpAG4AZwAgAD0AIAAnAGEAbgBvAHQAaABlAHIALQB0AGUAcwB0ACcACgAKAHMAdwBpAHQAYwBoACAAKABzAG8AbQBlAF8AcwB0AHIAaQBuAGcAKQAgAHsACgAgACAAIAAgAGMAYQBzAGUAIAAnAHQAZQBzAHQAJwA6AAoAIAAgACAAIAAgACAAIAAgAGIAcgBlAGEAawAKACAAIAAgACAAYwBhAHMAZQAgACcAYQBuAG8AdABoAGUAcgAtAHQAZQBzAHQAJwA6AAoAIAAgACAAIAAgACAAIAAgAGMAbwBuAHMAbwBsAGUALgBsAG8AZwAoACcAcwB1AGMAYwBlAHMAcwAnACkACgAgACAAIAAgACAAIAAgACAAYgByAGUAYQBrAAoAIAAgACAAIABkAGUAZgBhAHUAbAB0ADoACgAgACAAIAAgACAAIAAgACAAYgByAGUAYQBrAAoAfQAKAAoAcwB3AGkAdABjAGgAIAAoAE0AYQB0AGgALgByAG8AdQBuAGQAKABNAGEAdABoAC4AcgBhAG4AZABvAG0AKAApACAAKgAgADEAMAApACkAIAB7AAoAIAAgACAAIABjAGEAcwBlACAAMQA6AAoAIAAgACAAIAAgACAAIAAgAGIAcgBlAGEAawAKACAAIAAgACAAYwBhAHMAZQAgADIAOgAKACAAIAAgACAAIAAgACAAIABiAHIAZQBhAGsACgAgACAAIAAgAGMAYQBzAGUAIAAzADoACgAgACAAIAAgACAAIAAgACAAYgByAGUAYQBrAAoAIAAgACAAIABjAGEAcwBlACAANAA6AAoAIAAgACAAIAAgACAAIAAgAGIAcgBlAGEAawAKACAAIAAgACAAYwBhAHMAZQAgADUAOgAKACAAIAAgACAAIAAgACAAIABiAHIAZQBhAGsACgAgACAAIAAgAGMAYQBzAGUAIAA2ADoACgAgACAAIAAgACAAIAAgACAAYgByAGUAYQBrAAoAIAAgACAAIABjAGEAcwBlACAANwA6AAoAIAAgACAAIAAgACAAIAAgAGIAcgBlAGEAawAKACAAIAAgACAAYwBhAHMAZQAgADgAOgAKACAAIAAgACAAIAAgACAAIABiAHIAZQBhAGsACgAgACAAIAAgAGMAYQBzAGUAIAA5ADoACgAgACAAIAAgACAAIAAgACAAYgByAGUAYQBrAAoAIAAgACAAIABkAGUAZgBhAHUAbAB0ADoACgAgACAAIAAgACAAIAAgACAAYgByAGUAYQBrAAoAfQA%3D)

**Comment by ematipico** [comment]
@lewiswolf you didn't the rule to use from the dropdown in the left. Here's a [valid one ](https://biomejs.dev/playground/?lintRules=noUnnecessaryConditions&tab=syntax&pane=Console&code=YwBvAG4AcwB0ACAAcwBvAG0AZQBfAHMAdAByAGkAbgBnADoAIABzAHQAcgBpAG4AZwAgAD0AIAAnAGEAbgBvAHQAaABlAHIALQB0AGUAcwB0ACcACgAKAHMAdwBpAHQAYwBoACAAKABzAG8AbQBlAF8AcwB0AHIAaQBuAGcAKQAgAHsACgAgACAAIAAgAGMAYQBzAGUAIAAnAHQAZQBzAHQAJwA6AAoAIAAgACAAIAAgACAAIAAgAGIAcgBlAGEAawAKACAAIAAgACAAYwBhAHMAZQAgACcAYQBuAG8AdABoAGUAcgAtAHQAZQBzAHQAJwA6AAoAIAAgACAAIAAgACAAIAAgAGMAbwBuAHMAbwBsAGUALgBsAG8AZwAoACcAcwB1AGMAYwBlAHMAcwAnACkACgAgACAAIAAgACAAIAAgACAAYgByAGUAYQBrAAoAIAAgACAAIABkAGUAZgBhAHUAbAB0ADoACgAgACAAIAAgACAAIAAgACAAYgByAGUAYQBrAAoAfQAKAAoAcwB3AGkAdABjAGgAIAAoAE0AYQB0AGgALgByAG8AdQBuAGQAKABNAGEAdABoAC4AcgBhAG4AZABvAG0AKAApACAAKgAgADEAMAApACkAIAB7AAoAIAAgACAAIABjAGEAcwBlACAAMQA6AAoAIAAgACAAIAAgACAAIAAgAGIAcgBlAGEAawAKACAAIAAgACAAYwBhAHMAZQAgADIAOgAKACAAIAAgACAAIAAgACAAIABiAHIAZQBhAGsACgAgACAAIAAgAGMAYQBzAGUAIAAzADoACgAgACAAIAAgACAAIAAgACAAYgByAGUAYQBrAAoAIAAgACAAIABjAGEAcwBlACAANAA6AAoAIAAgACAAIAAgACAAIAAgAGIAcgBlAGEAawAKACAAIAAgACAAYwBhAHMAZQAgADUAOgAKACAAIAAgACAAIAAgACAAIABiAHIAZQBhAGsACgAgACAAIAAgAGMAYQBzAGUAIAA2ADoACgAgACAAIAAgACAAIAAgACAAYgByAGUAYQBrAAoAIAAgACAAIABjAGEAcwBlACAANwA6AAoAIAAgACAAIAAgACAAIAAgAGIAcgBlAGEAawAKACAAIAAgACAAYwBhAHMAZQAgADgAOgAKACAAIAAgACAAIAAgACAAIABiAHIAZQBhAGsACgAgACAAIAAgAGMAYQBzAGUAIAA5ADoACgAgACAAIAAgACAAIAAgACAAYgByAGUAYQBrAAoAIAAgACAAIABkAGUAZgBhAHUAbAB0ADoACgAgACAAIAAgACAAIAAgACAAYgByAGUAYQBrAAoAfQA%3D)

**Comment by lewiswolf** [comment]
sorry for the misunderstanding - thank you!
